modules/currency-l0/src/main/scala/io/constellationnetwork/currency/l0/http/routes/ValidationErrorRoutes.scala

```diff
@@ -0,0 +1,47 @@
+package io.constellationnetwork.currency.l0.http.routes
+
+import java.time.Instant
+
+import cats.effect.Async
+import cats.syntax.flatMap._
+
+import io.constellationnetwork.node.shared.domain.block.processing.BlockRejectionReason
+import io.constellationnetwork.node.shared.infrastructure.consensus.ValidationErrorStorage
+import io.constellationnetwork.node.shared.snapshot.currency
+import io.constellationnetwork.node.shared.snapshot.currency.CurrencySnapshotEvent
+import io.constellationnetwork.routes.internal._
+
+import eu.timepit.refined.auto._
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Json}
+import org.http4s.HttpRoutes
+import org.http4s.circe.CirceEntityEncoder._
+import org.http4s.dsl.Http4sDsl
+
+final case class ValidationErrorRoutes[F[_]: Async](
+  currencySnapshotValidationErrorStorage: ValidationErrorStorage[F, CurrencySnapshotEvent, Option[BlockRejectionReason]]
+) extends Http4sDsl[F]
+    with PublicRoutes[F] {
+
+  implicit val encoder: Encoder[(CurrencySnapshotEvent, Option[BlockRejectionReason], Instant)] =
+    new Encoder[(CurrencySnapshotEvent, Option[BlockRejectionReason], Instant)] {
+      override def apply(a: (CurrencySnapshotEvent, Option[BlockRejectionReason], Instant)): Json = Json.obj(
+        (
+          "block",
+          a._1 match {
+            case currency.BlockEvent(value)            => value.asJson
+            case currency.DataApplicationBlockEvent(_) => Json.Null
+            case currency.CurrencyMessageEvent(_)      => Json.Null
+          }
+        ),
+        ("timestamp", a._3.asJson)
+      )
+    }
+
+  override protected val prefixPath: InternalUrlPrefix = "/currency"
+
+  override protected val public: HttpRoutes[F] = HttpRoutes.of[F] {
+    case GET -> Root / "validation-errors" =>
```

Comment:
I am wondering if it shouldn't be a sub-resource of `/snapshot/events/validation-errors` but if it is currency-framework specific then maybe `currency/snapshot-event-validation-errors` or something more specific 🤔